File: 1923-sentence-similarity-iii/sentence-similarity-iii.py
class Solution:
    def areSentencesSimilar(self, sentence1: str, sentence2: str) -> bool:
        arr1 = sentence1.split(" ")
        arr2 = sentence2.split(" ")

        if (len(arr2) < len(arr1)):
            arr1, arr2 = arr2, arr1

        i = 0
        j = len(arr1) - 1
        k = len(arr2) - 1

        while (i < len(arr1) and arr1[i] == arr2[i]):
            i += 1

        while (j >= 0 and k >= 0 and arr1[j] == arr2[k]):
            j -= 1
            k -= 1

        return i > j
        

File: 1923-sentence-similarity-iii/sentence-similarity-iii.js
var areSentencesSimilar = function(sentence1, sentence2) {
    let arr1 = sentence1.split(" ")
    let arr2 = sentence2.split(" ")

    if (arr1.length > arr2.length) {
        [arr1, arr2] = [arr2, arr1]
    }

    let i = 0
    let j = arr1.length
    let k = arr2.length

    while (i < arr1.length && arr1[i] == arr2[i]) {
        i += 1
    }

    while (j >= 0 && k >= 0 && arr1[j] == arr2[k]) {
        j -= 1
        k -= 1
    }

    return i > j

};